CODING_GUIDELINES.md

```diff
@@ -25,6 +25,13 @@ We deviate from the [Google Python Style Guide][google-style-guide] only in the
 - We use [`black`][black] and [`isort`][isort] for source code and imports formatting, which may work differently than indicated by the guidelines in section [_3. Python Style Rules_](https://google.github.io/styleguide/pyguide.html#3-python-style-rules). For example, maximum line length is set to 100 instead of 79 (although docstring lines should still be limited to 79).
 - According to subsection [_2.19 Power Features_](https://google.github.io/styleguide/pyguide.html#219-power-features), direct use of _power features_ (e.g. custom metaclasses, import hacks, reflection) should be avoided, but standard library classes that internally use these power features are accepted. Following the same spirit, we allow the use of power features in infrastructure code with similar functionality and scope as the Python standard library.
 - According to subsection [_3.19.12 Imports For Typing_](https://google.github.io/styleguide/pyguide.html#31912-imports-for-typing), symbols from `typing` and `collections.abc` modules used in type annotations _"can be imported directly to keep common annotations concise and match standard typing practices"_. Following the same spirit, we allow symbols to be imported directly from third-party or internal modules when they only contain a collection of frequently used typying definitions.
+- Imports [_2.2 Imports_](https://google.github.io/styleguide/pyguide.html#22-imports):
+  - Importing individual classes and functions is allowed if they are to be used in DSL code and the DSL does not allow them to be qualified.
```

Comment:
Direct imports are required anyway for this use case, so this is not really a convention but the documentation of a required behavior, right? Additionally, it only affects user code (e.g. examples, tests, ...), so I would move this to the user documentation (either the tutorial or an appropriate docstring)